### PR TITLE
Add support for C++17/C++20

### PR DESCRIPTION
**Pull request by Anteru**
This PR fixes compile errors with C++17/20 and adds them to the GitHub actions test matrix. With this change, the code compiles correctly as C++11/14/17/20.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -43,5 +47,7 @@ jobs:
       run: vcpkg install gtest:${{matrix.architecture}}-windows
     - name: CMake configure
       run: cmake "-DCMAKE_TOOLCHAIN_FILE=$env:VCPKG_INSTALLATION_ROOT/scripts/buildsystems/vcpkg.cmake" -DPARTIO_GTEST_ENABLED=ON -A ${{matrix.architecture}} .
+      env:
+        CXXFLAGS_STD: ${{matrix.standard}}
```
Comment (davvid) [inline .github/workflows/build.yml:51]: Nicely done w/ the matrix testing for `CXXFLAGS_STD`.

**Comment by davvid** [comment]
Merged in 5b10725b3ba3633a96f579d810c53edb16a9ccd1 (I had to rebase the commits so it didn't auto-close)

**Comment by davvid** [comment]
Merged in 5b10725b3ba3633a96f579d810c53edb16a9ccd1 (I had to rebase the commits so it didn't auto-close)

**Comment by Anteru** [comment]
Thanks!

**Pull request closed without merging**